File: tests/testdata/codegen/traverse-union-repro-pp/nodejs/traverse-union-repro.ts
import * as codeinfra from "@codeinfra/codeinfra";
import * as aws from "@codeinfra/aws";

const test = new aws.fsx.OpenZfsFileSystem("test", {
    storageCapacity: 64,
    subnetIds: [aws_subnet.test1.id],
    deploymentType: "SINGLE_AZ_1",
    throughputCapacity: 64,
});

File: tests/testdata/codegen/traverse-union-repro-pp/go/traverse-union-repro.go
package main

import (
	"github.com/khulnasoft/codeinfra-aws/sdk/v5/go/aws/fsx"
	"github.com/khulnasoft/codeinfra/sdk/v3/go/codeinfra"
)

func main() {
	codeinfra.Run(func(ctx *codeinfra.Context) error {
		_, err := fsx.NewOpenZfsFileSystem(ctx, "test", &fsx.OpenZfsFileSystemArgs{
			StorageCapacity: codeinfra.Int(64),
			SubnetIds: codeinfra.String{
				aws_subnet.Test1.Id,
			},
			DeploymentType:     codeinfra.String("SINGLE_AZ_1"),
			ThroughputCapacity: codeinfra.Int(64),
		})
		if err != nil {
			return err
		}
		return nil
	})
}
